ISIC 2018 Improved UNET architecture

isak-ow opened the pull request:
This pullrequest contains my attemt at implementing the improved unet model proposed by the paper in the assignment. Because of the amount of jobs in queue for gpu on the rangpur cluster, I had to train the model in google colab (same code). After training the model it achieved a dice coefficient of 0.8 and an accuracy of 0.9 (may be a wierd metric). See the README.md for more information.

isak-ow committed:
Forked correctly and switched to topic branch

isak-ow committed:
added crude first attempt at unet model and functionality for fetching dataset

isak-ow committed:
model giving correct sizes

isak-ow committed:
added softmax layer to model

isak-ow committed:
added printing of metrics and dice similarity coefficient for training module

isak-ow committed:
added sh file for cluster

isak-ow committed:
added gitignore for dataset

isak-ow committed:
added gitignore for dataset

isak-ow committed:
changed sh file to run training script

isak-ow committed:
changed sh file to run training script

isak-ow committed:
fixed utility functions

isak-ow committed:
corrected accuracy function

isak-ow committed:
debugging

isak-ow committed:
fixed gitignore

isak-ow committed:
pulling data from cluster instead of google drive

isak-ow committed:
pulling data from cluster instead of google drive

isak-ow committed:
pulling data from cluster instead of google drive

isak-ow committed:
fixed size mismatch in training

isak-ow committed:
fixed size mismatch in training

isak-ow committed:
switched to Adam optimizer

isak-ow committed:
training with 6 epochs

isak-ow committed:
training with 6 epochs

isak-ow committed:
changed to dice loss function

isak-ow committed:
changed training script

isak-ow committed:
changed training script

isak-ow committed:
testing training script

isak-ow committed:
training on vgpu

isak-ow committed:
training on vgpu

isak-ow committed:
trying adam again

isak-ow committed:
test run with adam and lower lr

isak-ow committed:
trying complete the model

isak-ow committed:
fixing dataset module

Commit:
working training script

Commit:
testing other model to compare performance

Commit:
updated report

Commit:
working training script, running on colab

isak-ow committed:
added the images from colab

isak-ow committed:
added image to readme

isak-ow committed:
added image to readme

isak-ow committed:
completed training script with logging and changed learning rate to 0.001

Commit:
added comments and no_grad for validation

Commit:
added photo where dice coeff is 0.8

Commit:
added more images for

Commit:
added more images for readmefile

Commit:
added plots to readme

Commit:
added plots to readme

Commit:
added plots to readme

Commit:
added plots to readme

Commit:
added dependencies

Commit:
moved image

Commit:
moved image

Commit:
added links

Commit:
added links

Commit:
added links

Commit:
added links

wangzhaomxy commented:
This is an initial inspection, no action is required at this point.

- Well organized files, but wrong files path. The student constructed document should be packed in the ‘recognition’ document.
- Algorithm can solve the problem, but no model training evidence/record. The err.txt shows some training error occurred, and no out.txt content, both of which indicates a failed training attempt. 
- Try to use PyTorch to construct the Improved U-Net model, and the model body structure seems good. Some useless functions in utilities.py file. No data augmentation.
- Insufficient comment in the code, no docstring, also need some references.
- Easy difficulty problem, no higher difficulty marks.

isak-ow committed:
Opprettet med Colaboratory

isak-ow committed:
fixed file paths and added training script to show training

isak-ow committed:
Opprettet med Colaboratory

isak-ow committed:
added correct training file

isak-ow commented:
I changed the file path according to the feedback from Zhao, and added a training script from Colab. As mentioned in the Readme, the training I did had to be done on colab due to the queue on the rangpur cluster.

shakes76 commented:
# Marking
Marked based on content submitted before due date.
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, but shows error message -3
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem, no model training evidence -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting minimal -1
No Data leakage, no augmentation -1
Difficulty: Easy -10
## Commit Log
Meaningful commit messages minimal -1
Progressive commits used
## Documentation
ReadMe OK, no usage -1, no refs -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in InCorrect Branch) initial PR -2
Feedback required, log directories and pyc files need to be removed -2
Request Description good

shakes76 commented:
Will not merge due to log files commit and would make repo too large. Feedback marks possible +2 if the requested changes are made (see above), but no merge will be done.

isak-ow commented:
Hi,

In the initial inspection of the code I was told that some changes could be
made, but that no action was required for the project to pass. After
reading your grading it has me worried that I will not pass the course. I
will of course address the issues I can, raised in your comment, but Im
quite stressed that I will recieve a failing grade. Could you speak to this?

Kind regards,

Isak Orlando Wangensteen
studentID: 48201292



On Mon, 20 Nov 2023 at 12:53, Shakes ***@***.***> wrote:

> Will not merge due to log files commit and would make repo too large.
> Feedback marks possible +2 if the requested changes are made (see above),
> but no merge will be done.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/shakes76/PatternAnalysis-2023/pull/127#issuecomment-1818147289>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5JSL7KSHWLW2LDRLECO2T3YFLAZ3AVCNFSM6AAAAAA6JAUUN2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMJYGE2DOMRYHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Commit:
Removed log files and the pyc files as requested in the feedback

Commit:
added more comments explaining the code

Commit:
fixed the comments

Commit:
Added references

isak-ow commented:
Removed the log directories and the pyc files as requested in the feedback

wangzhaomxy commented:
Feedback attempt and feedback marks granted +2.
